src/main/java/org/jmxtrans/agent/influxdb/InfluxDbOutputWriter.java

```diff
@@ -0,0 +1,222 @@
+        tags = InfluxMetricConverter.tagsFromCommaSeparatedString(tagsStr);
+        connectTimeoutMillis = ConfigurationUtils.getInt(settings, "connectTimeoutMillis", 3000);
+        readTimeoutMillis = ConfigurationUtils.getInt(settings, "readTimeoutMillis", 5000);
+        url = parseUrlStr(getWriteEndpointForUrlStr(urlStr));
+        logger.log(getInfoLevel(), "InfluxDbOutputWriter is configured with url=" + urlStr
+                + ", database=" + database
+                + ", user=" + user
+                + ", password=" + (password != null ? "****" : null)
+                + ", tags=" + tagsStr
+                + ", connectTimeoutMills=" + connectTimeoutMillis
+                + ", readTimeoutMillis=" + readTimeoutMillis);
+    }
+
+    private String getWriteEndpointForUrlStr(String urlStr) {
+        return urlStr + (urlStr.endsWith("/") ? "write" : "/write");
+    }
+
+    private URL parseUrlStr(String urlStr) {
+        try {
+            return new URL(urlStr + "?" + buildQueryString());
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String buildQueryString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("precision=ms")
+                .append("&db=").append(database);
+        appendParamIfNotEmptyOrNull(sb, "u", user);
+        appendParamIfNotEmptyOrNull(sb, "p", password);
+        appendParamIfNotEmptyOrNull(sb, "rp", retentionPolicy);
+        return sb.toString();
+    }
+
+    @Override
+    public void writeInvocationResult(String invocationName, Object value) throws IOException {
+        writeQueryResult(invocationName, null, value);
+    }
+
+    @Override
+    public void writeQueryResult(String metricName, String metricType, Object value) throws IOException {
+        InfluxMetric metric = InfluxMetricConverter.convertToInfluxMetric(metricName, value, tags,
+                clock.getCurrentTimeMillis());
+        batchedMetrics.add(metric);
+    }
+
+    @Override
+    public void postCollect() throws IOException {
+        String body = convertMetricsToLines(batchedMetrics);
+        String queryString = buildQueryString();
+        if (logger.isLoggable(getTraceLevel())) {
+            logger.log(getTraceLevel(), "Sending to influx (" + url + "):\n" + body);
+        }
+        batchedMetrics.clear();
+        sendMetrics(queryString, body);
+    }
+
+    private void sendMetrics(String queryString, String body) throws IOException {
+        HttpURLConnection conn = createAndConfigureConnection();
+        writeMetrics(conn, body);
+        int responseCode = conn.getResponseCode();
+        if (responseCode / 100 != 2) {
+            throw new RuntimeException("Failed to write metrics, response code: " + responseCode
+                    + ", response message: " + conn.getResponseMessage());
+        }
+        String response = readResponse(conn);

```

Comment:
Would it make sense to `org.jmxtrans.agent.util.io.IoUtils#closeQuietly(java.net.URLConnection)` and ensure that the connection is returned to the http connection pool?